src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1227,6 +1227,17 @@ protected sealed override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownA
             {
                 arguments.GetOrCreateData<TypeWellKnownAttributeData>().HasCompilerLoweringPreserveAttribute = true;
             }
+            else if (attribute.IsTargetAttribute(AttributeDescription.ExtendedLayoutAttribute))
+            {
+                if (attribute.AttributeClass.ExtendedSpecialType != InternalSpecialType.System_Runtime_InteropServices_ExtendedLayoutAttribute)
+                {
+                    diagnostics.Add(ErrorCode.ERR_InvalidExtendedLayoutAttribute, arguments.AttributeSyntaxOpt.Name.Location);
+                }
```

Comment:
I would not expect this check. I would instead expect a check down on line 2001ish, like inline arrays, that ensures the target runtime supports extended layouts. That, in turn, will ensure that the right ctor exists.